rfcs/72-editor.md

```diff
@@ -0,0 +1,508 @@
+`target` folder.
+
+Alternatively, Bevy can publish a new build every time you push to a repository.
+Right now this is only possible using GitHub and Itch.io, sign in with `bevy
+login`:
+
+```
+$ bevy login github
+Enter GitHub access token: ***
+$ bevy login itch
+Enter Itch.io access token: ***
+```
+
+**Note that these commands store the tokens in you system vault, not within the
+repository.**
+
+### Scene and Entity Manipulator
+
+Bevy comes with an entity and scene manipulator. The easiest way to use these
+tools is with the graphical editor, run `bevy editor` in the project directory.
+
+The entity manipulator lists all of the entities in a given world, as well as
+the components that are associated with those entities. You can add new entities
+from this view, and attach new components to them. You can also edit the
+entities that were added using the manipulator.
+
+You can view all entities in your world, not just ones created with the
+manipulator. However, you are only guaranteed to be able to edit static
+entities, and entities created with the manipulator.
+
+In addition to the entity manipulator, Bevy also provides developers with a
+scene manipulator. This allows developers to place entities in a larger level,
+and get a real-time view of how they interact with each other.
+
+### Systems Manager
+
+In addition to the manipulator, Bevy provides developers with a view of all the
+systems their games uses. In addition to creating new systems, developers can
+use this manager to find and open the code for existing systems. The systems
+manager can also be viewed in the graphical editor, run `bevy editor` in the
+project directory.
+
+### Localization, Accessibility, Porting, and Plugins
+
+Just like the rest of the Bevy engine, the editor can be extended using plugins.
+Some plugins are installed to the system, while others are added only to a
+project. Some possible plugin features are:
+
+* Adding new sub-commands to the `bevy` CLI tool.
+* Adding new views to the graphical editor.
+* Adding context-specific renderings of component data.
+* Localize both the editor and underlying game.
+* Alter game behavior when porting to new platforms.
+
+## Implementation strategy
+
+The following is a list of features that are needed for the minimum viable
+editor:
+
+* Project/System separation
+* Version dispatching
+* CLI Interface
+* Window Interface
+* New project construction
+* Asset management
+* Distributable packaging
+* Credential/secrets management
+* Scene/Entity manipulator (with possible component editing dispatch).
+* Systems list
+* Plugin architecture
+
+While it is possible to make a prototype with a subset of these features, each
+feature implementation must (1) makes space for every other feature, and can (2)
+assumes the other features exist.
+
+The rest of this section will be a sketch out the rough implementation for each
+of these features.
+
+### Project/System separation
```

Comment:
We should probably mention that "these crates live in the bevyengine/bevy repo" somewhere in this section.